src/main/java/emu/grasscutter/server/packet/send/PacketResinChangeNotify.java

```diff
@@ -5,19 +5,17 @@
 import emu.grasscutter.net.packet.BasePacket;
 import emu.grasscutter.net.packet.PacketOpcodes;
 import emu.grasscutter.net.proto.ResinChangeNotifyOuterClass.ResinChangeNotify;
+import lombok.val;
 
 public class PacketResinChangeNotify extends BasePacket {
-	
-	public PacketResinChangeNotify(Player player) {
-		super(PacketOpcodes.ResinChangeNotify);
-		
-		ResinChangeNotify proto = ResinChangeNotify.newBuilder()
-				.setCurValue(player.getProperty(PlayerProperty.PROP_PLAYER_RESIN))
-				.setNextAddTimestamp(player.getNextResinRefresh())
-				.build();
+    public PacketResinChangeNotify(Player player) {
+        super(PacketOpcodes.ResinChangeNotify);
 
-		// ToDo: Add ability to buy resin with primogems, has to be included here.
-		
-		this.setData(proto);
-	}
+        val proto = ResinChangeNotify.newBuilder()
+            .setCurValue(player.getProperty(PlayerProperty.PROP_PLAYER_RESIN))
+            .setNextAddTimestamp(player.getNextResinRefresh())
+            .setCurBuyCount(player.getResinBuyCount());
+
```

Comment:
missing `.build()` here.

Reply:
You don't need to build them, since there is a method that accepts the builder and calls build itself. So it works this way too and is also less code.
Also it would probably failed to build if it was required ^^